analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicContext.cs

```diff
@@ -43,5 +43,14 @@ public SymbolicContext(IOperationWrapperSonar operation, ProgramState state)
 
         public SymbolicContext WithState(ProgramState newState) =>
             State == newState ? this : new(Operation, newState);
+
+        public bool HasConstraint(SymbolicConstraint constraint) =>
+            HasConstraint(constraint, Operation);
+
+        public bool HasConstraint(SymbolicConstraint constraint, IOperationWrapperSonar operation) =>
+            HasConstraint(constraint, operation.Instance);
+
+        public bool HasConstraint(SymbolicConstraint constraint, IOperation operation) =>
+            State[operation]?.HasConstraint(constraint) is true;
```

Comment:
Maybe it is better to move these "helpers" into some extension methods class.

Reply:
No, that would just scatter the logic over multiple places. We can modify this class, so we don't need extensions.